drivers/sensor/ism330dhcx/ism330dhcx_shub.c

```diff
@@ -419,10 +419,11 @@ static struct ism330dhcx_shub_slist {
 
 static inline void ism330dhcx_shub_wait_completed(struct ism330dhcx_data *data)
 {
-	u16_t freq;
+	ism330dhcx_status_master_t status;
 
-	freq = (data->accel_freq == 0) ? 26 : data->accel_freq;
-	k_sleep((2000U / freq) + 1);
+	do {
+		ism330dhcx_sh_status_get(data->ctx, &status);
```

Comment:
are you sure you want to busy-wait instead of sleeping here? how long does this operation typically take?

Reply:
maybe a couple of msec.
I can add a k_msleep(1), so that it put the thread to sleep after each iteration.
Would you agree?

Reply:
But I would keep the status reading check to know when op is done.

Reply:
Yes

Reply:
done, thx